worker_test.go

```diff
@@ -2,21 +2,344 @@ package main
+			BasicConstraintsValid: true,
+
+			IsCA: true,
+		}
+		caBody, err := x509.CreateCertificate(rand.Reader, &caCert, &caCert, caKey.Public(), caKey)
+		if err != nil {
+			m.ca.err = fmt.Errorf("unable to generate CA certificate: %v", err)
+			return
+		}
+		caReparsed, err := x509.ParseCertificate(caBody)
+		if err != nil {
+			m.ca.err = fmt.Errorf("unable to parse CA certificate: %v", err)
+			return
+		}
+		m.ca.roots = x509.NewCertPool()
+		m.ca.roots.AddCert(caReparsed)
+
+		chainKey, err := rsa.GenerateKey(rand.Reader, 2048)
+		if err != nil {
+			m.ca.err = fmt.Errorf("unable to generate sub-CA key: %v", err)
+			return
+		}
+		chainCert := x509.Certificate{
+			SerialNumber: big.NewInt(2),
+			Subject: pkix.Name{
+				Organization: []string{"Testing Sub-CA"},
+			},
+			NotBefore: time.Time{},
+			NotAfter:  time.Now().Add(tenYears),
+
+			KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+			BasicConstraintsValid: true,
+
+			IsCA: true,
+		}
+		chainBody, err := x509.CreateCertificate(rand.Reader, &chainCert, caReparsed, chainKey.Public(), caKey)
+		if err != nil {
+			m.ca.err = fmt.Errorf("unable to generate sub-CA certificate: %v", err)
+			return
+		}
+		chainReparsed, err := x509.ParseCertificate(chainBody)
+		if err != nil {
+			m.ca.err = fmt.Errorf("unable to parse sub-CA certificate: %v", err)
+			return
+		}
+
+		m.ca.chainKey = chainKey
+		m.ca.chainCert = chainReparsed
+	})
+
+	certKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		require.NoErrorf(m.t, err, "unable to generate certificate key")
+	}
+	cert := x509.Certificate{
+		SerialNumber: big.NewInt(3),
+		DNSNames:     altNames,
+		NotBefore:    notBefore,
+		NotAfter:     notAfter,
+
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+
+	body, err := x509.CreateCertificate(rand.Reader, &cert, m.ca.chainCert, certKey.Public(), m.ca.chainKey)
+	if err != nil {
+		require.NoErrorf(m.t, err, "unable to generate certificate")
+	}
+	reparsed, err := x509.ParseCertificate(body)
+	if err != nil {
+		require.NoErrorf(m.t, err, "unable to parse certificate")
+	}
+
+	c := certs.NewCertificate(arn, reparsed, []*x509.Certificate{m.ca.chainCert})
+	return []*certs.CertificateSummary{c.WithRoots(m.ca.roots)}, nil
+}
+
+func TestResourceConversion(tt *testing.T) {
+	clusterIDTagPrefix := "kubernetes.io/cluster/"
+	clusterID := "aws:123:eu-central-1:kube-1"
+	vpcID := "1"
+	securityGroupID := "42"
+	running := int64(16) // See https://github.com/aws/aws-sdk-go/blob/master/service/ec2/api.go, type InstanceState
+
+	for _, scenario := range []struct {
+		name           string
+		ec2Responses   fake.Ec2MockOutputs
+		asgResponses   fake.AutoscalingMockOutputs
+		elbv2Responses fake.Elbv2MockOutputs
+		cfResponses    fake.CfMockOutputs
```

Comment:
nitpick on the naming:

1. In Go we should aim to use consistent case for initialisms and acronyms. So it should be `EC2` instead of `Ec2`. Same with ELB and CF: https://github.com/golang/go/wiki/CodeReviewComments#initialisms
2. Since the package is already called `fake` the actual types may not need to be called something with `Mock`. E.g. `fake.CFOutputs` would be clear IMO.


Number 1 I would generally recommend to follow, 2 is more of a opinion so feel free to ignore that. :)